aws/lambda/log-classifier/src/engine.rs

```diff
@@ -44,3 +45,381 @@ pub fn evaluate_ruleset(ruleset: &RuleSet, log: &Log) -> Option<Match> {
         .flat_map(|rule| evaluate_rule(rule, log))
         .max_by(|a, b| a.rule.priority.cmp(&b.rule.priority))
 }
+
+/// Evaluates a ruleset against all lines of a log, returning matching line numbers.
+///
+/// # Arguments
+///
+/// * `ruleset` - A reference to the RuleSet to evaluate.
+/// * `log` - A reference to the Log to evaluate against.
+///
+/// # Returns
+///
+/// A vector of line numbers that match any rule in the ruleset.
+/// Note: Each rule in ruleset can only match 1 line which is the last line that matches the rule.
+pub fn evaluate_ruleset_all_lines(ruleset: &RuleSet, log: &Log) -> Vec<usize> {
```

Comment:
Should this returns `Vec<Match>` like `evaluate_ruleset` instead of `Vec<usize>`?

Reply:
Either works, I think in practice it doesn't matter too much.

The information we actually care about in match is the text in the log, and the way that it is conveyed is the log + the line number. The rest of the info in match is not useful.